### PR TITLE
Minio exporter

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Export metrics from minio servers

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/monitoring/prometheus/minio-exporter.nix [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        default = 9290;
+        description = ''
+          Port to listen on.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "0.0.0.0";
```
Comment (Mic92) [inline nixos/modules/services/monitoring/prometheus/minio-exporter.nix]: No default value here?

File: nixos/modules/services/monitoring/prometheus/minio-exporter.nix [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        description = ''
+          Address to listen on for web interface and telemetry.
+        '';
+      };
+
+      minioAddress = mkOption {
+        type = types.str;
+        example = "https://10.0.0.1:9000";
+        description = ''
```
Comment (Mic92) [inline nixos/modules/services/monitoring/prometheus/minio-exporter.nix]: If we would have a minio service, this could default to the local instance.

Comment (bachp) [reply nixos/modules/services/monitoring/prometheus/minio-exporter.nix]: Good idea

Comment (bachp) [reply nixos/modules/services/monitoring/prometheus/minio-exporter.nix]: Done, it now also takes the Keys from the minio configuration.

File: nixos/modules/services/monitoring/prometheus/minio-exporter.nix [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        type = types.str;
+        example = "BKIKJAA5BMMU2RHO6IBB";
+        description = ''
+          The value of the Minio access key.
+          It is required in order to connect to the server.
+          By default this uses the one from the local minio server if enabled
+          and <literal>config.services.minio.accessKey</literal>.
+        '';
+      } // optionalAttrs (config.services.minio.enable && config.services.minio.accessKey != "") {
+        default = config.services.minio.accessKey;
```
Comment (Mic92) [inline nixos/modules/services/monitoring/prometheus/minio-exporter.nix]: by default minio keys are generated on startup.

Comment (bachp) [reply nixos/modules/services/monitoring/prometheus/minio-exporter.nix]: Oh you are right, I forgot about that. Thanks for fixing it.

**Pull request merged**